### PR TITLE
Fix issue #10949

### PR DESCRIPTION
**Pull request by markuspf**
Do not use 0 as a guard value to reconstruct a path

### CODE REVIEW

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/8235/

File: arangod/Graph/KShortestPathsFinder.cpp [hunk 1]
```diff
@@ -249,9 +249,9 @@ void KShortestPathsFinder::reconstructPath(Ball const& left, Ball const& right,
   DijkstraInfo* it;
   it = left._frontier.find(join);
   TRI_ASSERT(it != nullptr);
-  double startToJoin = it->_weight;
+  double startToJoin = it->weight();
```
Comment (mchacki) [inline arangod/Graph/KShortestPathsFinder.cpp]: I like if we do not look into private member :+1:

File: tests/js/server/aql/aql-graph-k-shortest-path-0-weight-regression.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    _key: "0"
+  });
+  vertices.push({
+    _key: "1"
+  });
+  vertices.push({
+    _key: "2"
+  });
+
+  edges.push({ _from: `${vName}/0`, _to: `${vName}/1`, weight: 0 });
```
Comment (mchacki) [inline tests/js/server/aql/aql-graph-k-shortest-path-0-weight-regression.js]: can you add another test, where weight: 0 is somewhere within the path?

eg A -1> B -0> C -1> D
Path A to D.

and a test with more than one path of different length (e.g. what you have + my suggestion)

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/8256/

**Comment by markuspf** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/8282/

**Pull request merged**